router/validation.go

```
@@ -160,18 +178,227 @@ func validateSearchRequest(r *http.Request) error {
+			// Validate the query structure
+			if err := validateQueryClause(queryMap); err != nil {
+				return err
 			}
-			return fmt.Errorf("invalid JSON: %v", err)
 		}
 	}
 
 	return nil
 }
+
+type validationError struct {
```

golangci-lint (1.62.2)

type `validationError` is unused

(unused)